model/task_queue.go

```diff
@@ -535,12 +535,12 @@ outer:
 
 	// validate that the task is there
 	if !found {
-		return errors.Errorf("task id '%s' was not found in the in-memory queue for distro '%s'",
-			taskId, self.Distro)
+		return errors.Errorf("TaskQueueItem with id '%s' not found in the in-memory TaskQueue.Queue for distro '%s'", taskId, self.Distro)
 	}
 
 	var err error
 	err = dequeue(taskId, self.Distro)
+	// STU: in what cases would there be a TaskQueueItem in the in-memory TaskQueue.Queue but no corresponding reference (Queue._id) in the db.task_queue?
```

Comment:
not sure, a race between multiple hosts? or some weird case where we manually drop the queues? are we confident that callers can handle this case correctly.

Reply:
This will happen a lot. When something is dequeued from the in-memory queue on one app server, it will still be present in every other app server's in-memory queue. It will only no longer be present after the TTL has passed, and each app server has re-created its in-memory queue.

Reply:
As discussed 1:1 -- thanks for confirming this is the case.